### DISCUSSION TITLE
Access to raw.githubusercontent.com

### DISCUSSION THREAD

**Discussion by jvidals** [ChristopherKlay/StadiaEnhanced#138] [General]
I have sensitive git repos up often for work with different accounts, and the fact that the newest update requests access to modify raw.githubusercontent.com makes me really nervous and I disabled the extension. Is this permission necessary?

**Comment by ChristopherKlay** [comment]
The permission is needed to be able to load the current database file (used to store the maximum resolution, fps and other infos about games) from this repository, instead of delivering it via the extension itself (which would require updates to the extension on a daily basis), while at the same time making it available to be viewed together with the source of the extension.

I can understand some people being nervous due to the permission showing that it can read raw data files from GitHub, however the manifest file of the extension is actually limiting the permission to this specific repository only.
```
"permissions": [
    "*://raw.githubusercontent.com/ChristopherKlay/StadiaEnhanced/*"
]
```
The content script also only loads on the Stadia website (and has no permissions for any other site) - it can't access your private data when viewing GitHub by any means, for example.

If you feel like this still gives the extension too much access, or you simply don't want it to be able to load the database to begin with, you can rightclick the extension, hit "Manage extension" and "Permissions" to specifically just disable the permission for GitHub. It *should* result in the extension working just fine, but any functionality linked to the database (i.e. being able to view the maximum resolution of a game) will be missing.

**Reply by jvidals** [reply]
Thank you for the reply!

**Reply by ChristopherKlay** [reply]
A [update](https://www.reddit.com/r/Stadia/comments/o1775e/stadia_enhanced_github_permission/) to this is available.

The newest version introduces a background script to manage file access, dropping the GitHub permissions.
